rustls/src/client/client_conn.rs

```diff
@@ -226,9 +216,7 @@ pub enum Tls12Resumption {
 impl Clone for ClientConfig {
     fn clone(&self) -> Self {
         Self {
-            cipher_suites: self.cipher_suites.clone(),
-            kx_groups: self.kx_groups.clone(),
-            provider: self.provider,
+            provider: Arc::<CryptoProvider>::clone(&self.provider),
```

Comment:
Do we really need the turbofish here? That seems surprising to me. If so maybe just prefer `self.provider.clone()` instead?

Reply:
IIRC `clippy` was flagging this without the turbofish but I will verify.

Reply:
Yeah:
```
error: using `.clone()` on a ref-counted pointer
   --> rustls/src/client/client_conn.rs:221:23
    |
221 |             provider: self.provider.clone(),
    |                       ^^^^^^^^^^^^^^^^^^^^^ help: try: `Arc::<crypto::CryptoProvider>::clone(&self.provider)`
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_ref_ptr
note: the lint level is defined here
   --> rustls/src/lib.rs:278:5
    |
278 |     clippy::clone_on_ref_ptr,
    |     ^^^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `rustls` (lib) due to previous error
```

Reply:
Yeah... I don't care much for that particular lint, but okay.